config/initializers/docker.rb

```diff
@@ -1,16 +1,14 @@
 require 'docker'
 
-if !Rails.env.test? && !ENV['PRECOMPILE']
-  if ENV['DOCKER_URL'].present?
-    Docker.url = ENV['DOCKER_URL']
-    Docker.options = { read_timeout: 600 }
-
-    # Confirm the Docker daemon is a recent enough version
-    Docker.validate_version!
+if !Rails.env.test? && !ENV['PRECOMPILE'] && ENV['DOCKER_FEATURE']
+  if ENV['DOCKER_REGISTRY'].blank?
+    puts '*** DOCKER_REGISTRY environment variable must be configured when DOCKER_FEATURE is enabled ***'
+    exit 1
   end
 
-  if ENV['DOCKER_FEATURE'] && !ENV['DOCKER_REGISTRY'].present?
-    puts '*** DOCKER_REGISTRY environment variable must be configured when DOCKER_FEATURE is enabled ***'
-    exit(1)
+  if (url = ENV['DOCKER_URL']).presence
+    Docker.url = url
+    Docker.options = { read_timeout: 600, connect_timeout: 2 }
+    Docker.validate_version! # Confirm the Docker daemon is a recent enough version
```

Comment:
validate even if DOCKER_URL is not set. (scenario where using a unix socket instead of a TCP connection)